csrc/flash_attn/src/fmha_dgrad_kernel_1xN_loop.h

```diff
@@ -31,7 +31,7 @@ inline __device__ void dot_do_o(const uint4 (&do_)[M], const uint4 (&o)[M], cons
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-template<typename Kernel_traits, bool Is_dropout, bool Is_causal, bool Is_first, bool Is_last, typename Params, typename Prng>
+template<typename Kernel_traits, bool Is_dropout, bool Is_causal, bool has_attn, bool has_bias, bool Is_first, bool Is_last, typename Params, typename Prng>
```

Comment:
rename all `has_attn` to `has_attn_mask`. and `has_bias` to `has_attn_bias` ?

Reply:
done